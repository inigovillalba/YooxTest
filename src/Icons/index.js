export { default as Audio } from "./Audio/Audio";
export { default as Delivery } from "./Delivery/Delivery";
export { default as Edit } from "./Edit/Edit";
export { default as Favourites } from "./Favourites/Favourites";
export { default as GiftCards } from "./GiftCards/GiftCards";
