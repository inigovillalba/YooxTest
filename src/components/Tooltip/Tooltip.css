.tooltip {
  display: none;
  position: relative;
}

.tooltip.active {
  display: block;
}

.tooltip ul {
  background: #000000;
  list-style: none;
  text-align: left;
  border-radius: 2px;
  font-size: 0.8em;
  letter-spacing: 0.04em;
  padding: 0;
  margin-top: 0;
  -webkit-box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.5);
  -moz-box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.5);
  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.5);
}

.pointer {
  margin-top: 10px;
  width: 0;
  height: 0;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-bottom: 10px solid #000000;
  position: relative;
  left: 80%;
}
