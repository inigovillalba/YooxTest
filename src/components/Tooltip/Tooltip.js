import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import TooltipItem from "../TooltipItem/TooltipItem";

import "./Tooltip.css";

class Tooltip extends Component {
  static propTypes = {
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    active: PropTypes.bool.isRequired
  };

  render() {
    const tooltipClass = classNames({
      tooltip: true,
      active: this.props.active
    });

    const { items } = this.props;

    return (
      <div className={tooltipClass}>
        <div className="pointer" />
        <ul>
          {items.map(item => {
            return (
              <TooltipItem key={item.id}>
                {item.icon && <item.icon />}
                <span>{item.text}</span>
              </TooltipItem>
            );
          })}
        </ul>
      </div>
    );
  }
}

export default Tooltip;
