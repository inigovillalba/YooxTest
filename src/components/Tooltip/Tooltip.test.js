/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import Tooltip from "./Tooltip";
import { Audio, Delivery, Edit, Favourites, GiftCards } from "../../Icons";

describe("Tooltip", () => {
  test("renders correctly", () => {
    const renderer = new ShallowRenderer();
    const tooltipItems = [
      { id: 0, icon: Audio, text: "Button Call To Action Longer" },
      { id: 1, icon: Delivery, text: "Button Call To Action Longer" },
      { id: 2, icon: Edit, text: "Button Call To Action Longer" },
      { id: 3, icon: Favourites, text: "Button Call To Action Longer" },
      { id: 4, icon: GiftCards, text: "Button Call To Action Longer" }
    ];

    const wrapper = renderer.render(
      <Tooltip active={false} items={tooltipItems} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
