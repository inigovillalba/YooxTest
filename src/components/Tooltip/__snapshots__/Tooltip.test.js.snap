// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<div
  className="tooltip"
>
  <div
    className="pointer"
  />
  <ul>
    <TooltipItem>
      <Audio
        fillColor="#ffffff"
        height={20}
        width={20}
      />
      <span>
        Button Call To Action Longer
      </span>
    </TooltipItem>
    <TooltipItem>
      <Delivery
        fillColor="#ffffff"
        height={20}
        width={20}
      />
      <span>
        Button Call To Action Longer
      </span>
    </TooltipItem>
    <TooltipItem>
      <Edit
        fillColor="#ffffff"
        height={20}
        width={20}
      />
      <span>
        Button Call To Action Longer
      </span>
    </TooltipItem>
    <TooltipItem>
      <Favourites
        fillColor="#ffffff"
        height={20}
        width={20}
      />
      <span>
        Button Call To Action Longer
      </span>
    </TooltipItem>
    <TooltipItem>
      <GiftCards
        fillColor="#ffffff"
        height={20}
        width={20}
      />
      <span>
        Button Call To Action Longer
      </span>
    </TooltipItem>
  </ul>
</div>
`;
